core/src/replay_stage.rs

```diff
@@ -1855,6 +1866,33 @@ impl ReplayStage {
         }
     }
 
+    fn cache_block_times(
+        blockstore: &Arc<Blockstore>,
+        bank_forks: &Arc<RwLock<BankForks>>,
+        rooted_slots: &[Slot],
+        cache_block_time_sender: &Option<CacheBlockTimeSender>,
+    ) {
+        if let Some(cache_block_time_sender) = cache_block_time_sender {
+            for slot in rooted_slots {
+                if blockstore
+                    .get_block_time(*slot)
+                    .unwrap_or_default()
+                    .is_none()
+                {
+                    let rooted_bank = bank_forks
+                        .read()
+                        .unwrap()
+                        .get(*slot)
+                        .expect("rooted slot doesn't exist")
```

Comment:
Let's gracefully fail instead of panicking.  We've been bitten by this before

Reply:
Ah sure. You want I should change this one as well? https://github.com/solana-labs/solana/blob/ec4453f87a8ba2d7d3983713c5a26ffa8784267e/core/src/replay_stage.rs#L1024 That's the logic I referred to.

Reply:
Nm, that's a separate question; I won't complicate this pr